enumerable.rb

```diff
@@ -0,0 +1,186 @@
+module Enumerable
+  def my_each
+    if self.class == Range
+      l = to_a
+      count = 0
+      while count < l.length
+        yield(l[count])
+        count += 1
+      end
+      l
+    elsif self.class == Hash
+      count = 0
+      while count < length
+        yield(self[count])
+        count += 1
+      end
+      self
+    else
+      count = 0
+      while count < length
+        yield(self[count])
+        count += 1
+      end
+      self
+    end
+  end
+
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+    self
+  end
+
+  def my_select
```

Comment:
* [ ] should have working `my_select` **(covered by automated tests)** identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

```[1,2,3].my_select != [1,2,3].select```